src/albums/dto/create-album.dto.ts

```diff
@@ -0,0 +1,25 @@
+import { ApiProperty } from '@nestjs/swagger';
+import {
+  IsNotEmpty,
+  IsNumber,
+  IsOptional,
+  IsString,
+  Min,
+} from 'class-validator';
+
+export class CreateAlbumDto {
+  @ApiProperty({ example: 'Innuendo' })
+  @IsString()
+  @IsNotEmpty()
+  name: string;
+
+  @ApiProperty({ example: '1991' })
+  @IsNumber()
+  @Min(0)
+  year: number;
+
+  @ApiProperty({ nullable: true, format: 'UUID' })
+  @IsString()
+  @IsOptional()
+  artistId: string | null; // refers to Artist
```

Comment:
IsUUID decorator of "class-validator" could be used to validate if "artistId" is a valid UUID (optionally accepts uuid version)

Reply:
Yeah, good point. I thought we can't use Is Optional and IsUUID together.